### PR TITLE
[feat] add apis to support SMP for guest VM

### PR DESCRIPTION
**Pull request by hky1999**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        rust-toolchain: [nightly-2024-05-02, nightly-2024-08-24]
+        rust-toolchain: [nightly, nightly-2024-05-02]
```
Comment (aarkegz) [inline .github/workflows/ci.yml:11]: Why should we change this?

File: src/lib.rs [hunk 2]
```diff
@@ -16,7 +16,7 @@ mod vcpu;
 
 pub use self::pcpu::Aarch64PerCpu;
 pub use self::vcpu::Aarch64VCpu;
-pub use vcpu::AxArchVCpuConfig;
+pub use self::vcpu::Aarch64VCpuCreateConfig;
```
Comment (aarkegz) [inline src/lib.rs:19]: Maybe merge these two `use` statement

**Review by aarkegz — DISMISSED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (hky1999) [reply .github/workflows/ci.yml:11]: I think that by using the latest nightly toolchain we can keep up with the newest modification brought by rust community

Comment (hky1999) [reply .github/workflows/ci.yml:11]: errors have been bypassed by `continue-on-error`

File: src/lib.rs (hunk 2 above)
Comment (hky1999) [reply src/lib.rs:19]: ok

File: .github/workflows/ci.yml (hunk 1 above)
Comment (aarkegz) [reply .github/workflows/ci.yml:11]: It's weird as `naked_asm!` in newer versions has already messed thing up once and we've decided to stick to older versions for a while until edition 2024 is released.

Comment (aarkegz) [reply .github/workflows/ci.yml:11]: If we use `continue-on-error`, it means the "newest modification brought by rust community" was not used by us at all.

Comment (hky1999) [reply .github/workflows/ci.yml:11]: > If we use `continue-on-error`, it means the "newest modification brought by rust community" was not used by us at al

but we can catch up with which part of our code is compatible with the  "newest modification brought by rust community"

Comment (aarkegz) [reply .github/workflows/ci.yml:11]: > > If we use `continue-on-error`, it means the "newest modification brought by rust community" was not used by us at al
> 
> but we can catch up with which part of our code is compatible with the "newest modification brought by rust community"

OK, this is a good enough reason.

**Review by aarkegz — APPROVED** [review]
LGTM

**Pull request merged**